recipes/meson/all/test_package/conanfile.py

```diff
@@ -7,13 +7,16 @@ class TestPackageConan(ConanFile):
     generators = "pkg_config"
 
     def build(self):
-        meson = Meson(self)
-        meson.configure(build_folder="build")
-        meson.build()
+        if not tools.cross_building(self):
+            with tools.RunEnvironment(self):
+                meson = Meson(self)
+                meson.configure(build_folder="build")
+                meson.build()
```

Comment:
```suggestion
        with tools.environment_append(tools.RunEnvironment(self).vars):
            meson = Meson(self)
            meson.configure(build_folder="build")
            meson.build()
```

Reply:
I think the `if not tools.cross_building` is needed because it is possible that the user has a python interpreter in his host profile.
meson is platform independent, but the interpreter might not be.

Reply:
as explained above, in cross-compilation, conan's meson build helper cannot work anyway, see https://github.com/conan-io/conan-center-index/pull/7360 for example